docs/source/getting-started/tabular-data/grouping.ipynb

```diff
@@ -387,7 +387,93 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "context = dp.Context.compositor(\n",
+    "    # Many columns contain mixtures of strings and numbers and cannot be parsed as floats,\n",
+    "    # so we'll set `ignore_errors` to true to avoid conversion errors.\n",
+    "    data=pl.scan_csv(\"sample_FR_LFS.csv\", ignore_errors=True),\n",
+    "    privacy_unit=dp.unit_of(contributions=36),\n",
+    "    privacy_loss=dp.loss_of(epsilon=1.0 / 4),\n",
+    "    # allow for one query\n",
+    "    split_evenly_over=1,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "For example, you can reuse the stable grouping keys released in the previous query:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div><style>\n",
+       ".dataframe > thead > tr,\n",
+       ".dataframe > tbody > tr {\n",
+       "  text-align: right;\n",
+       "  white-space: pre-wrap;\n",
+       "}\n",
+       "</style>\n",
+       "<small>shape: (1, 4)</small><table border=\"1\" class=\"dataframe\"><thead><tr><th>column</th><th>aggregate</th><th>distribution</th><th>scale</th></tr><tr><td>str</td><td>str</td><td>str</td><td>f64</td></tr></thead><tbody><tr><td>&quot;len&quot;</td><td>&quot;Frame Length&quot;</td><td>&quot;Integer Laplace&quot;</td><td>144.0</td></tr></tbody></table></div>"
+      ],
+      "text/plain": [
+       "shape: (1, 4)\n",
+       "┌────────┬──────────────┬─────────────────┬───────┐\n",
+       "│ column ┆ aggregate    ┆ distribution    ┆ scale │\n",
+       "│ ---    ┆ ---          ┆ ---             ┆ ---   │\n",
+       "│ str    ┆ str          ┆ str             ┆ f64   │\n",
+       "╞════════╪══════════════╪═════════════════╪═══════╡\n",
+       "│ len    ┆ Frame Length ┆ Integer Laplace ┆ 144.0 │\n",
+       "└────────┴──────────────┴─────────────────┴───────┘"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "query_age_ilostat = (\n",
+    "    context.query()\n",
+    "    .group_by(\"AGE\", \"ILOSTAT\")\n",
+    "    .agg(dp.len())\n",
+    "    .with_keys(df[\"AGE\", \"ILOSTAT\"])\n",
+    ")\n",
+    "\n",
+    "query_age_ilostat.summarize()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This query works by conducting a left join where the left dataset is your explicit list of grouping keys and the right dataset contains the results of the grouped aggregation.\n",
+    "Grouping keys that don't exist in the original data are imputed as if the differentially private statistic were computed on an empty data partition."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Invariant Group Keys\n",
```

Comment:
Paragraph below doesn't use the term "invariant", so it feels like there's a disconnect.